crates/js-sys/tests/wasm/JsString.rs

```diff
@@ -261,3 +262,10 @@ fn value_of() {
     let greeting = JsString::from("Hello world!");
     assert_eq!(greeting.value_of(), "Hello world!");
 }
+
+#[wasm_bindgen_test]
+fn js_string_inheritance() {
+    let string = JsString::from("Hello world!");
+    assert!(string.is_instance_of::<JsString>());
```

Comment:
I have added extend for String type but test still don't pass. What i 'm missing ?

Reply:
Similar to `Number`, JS has both string values and `String` objects:

![strings](https://user-images.githubusercontent.com/74571/44055108-375a0e6a-9ef9-11e8-9b90-f2c02ff34f2f.png)

Despite string values not being an object, `String.prototype == Object.getPrototypeOf("")` is true and all the prototype methods exist on string values.

As far as moving forward with this PR goes, we can have a `JsString.js` file that exports a function that constructs a new `String` object:

```js
exports.new_string_object = () => new String("hi");
```

and import that in the `JsString.rs` test, and use that for the `is_instance_of` test.

Reply:
Whoa ! Good to know. Thank for the reply.